conftest.py

```diff
@@ -27,3 +29,18 @@ def pytest_runtest_setup(item):
     if envnames:
         if item.config.getoption("-E") not in envnames:
             pytest.skip("test requires env in %r" % envnames)
+
+
+@pytest.fixture(autouse=True)
```

Comment:
Would it be possible to not use `autouse` here, so that the fixture is explicitly used in test_use_stdin? My preference would be to avoid pytest magic if possible.

Reply:
Since `lsf_version()` is called by default (unless `use-stdin` is specified in the config), then this monkey-patch is needed by all tests that instantiate `LSFCluster()`.  That includes every test in `test_lsf.py` and also half of the tests in `test_jobqueue_core.py`.

I'm not a `pytest` expert, but IIUC, we need to use `autouse=True` or we need to add this fixture to every test that needs it in those two files.  Is there some better mechanism I'm missing?

Reply:
Right I missed that. I think we can keep it like this for this PR.

When we switch to use_stdin=True, we should remove the `lsid` logic (and so this autouse fixture). Basically, we thought there was a change in behaviour linked to LSF 10 and my current understanding is that this is not the case but is linked to some quirks on Summit ...